crates/engine/tree/src/persistence.rs

```diff
@@ -148,23 +148,31 @@ where
             match action {
                 PersistenceAction::RemoveBlocksAbove((new_tip_num, sender)) => {
                     let output = self.remove_blocks_above(new_tip_num);
-                    sender.send(output).unwrap();
+
+                    // we ignore the error because the caller may or may not care about the result
+                    let _ = sender.send(output);
```

Comment:
Maybe too early to focus on this, but could we log the potential error here?

Reply:
The error returned here is not actually an error, instead it's the item itself if sending is not possible:
https://docs.rs/tokio/latest/tokio/sync/oneshot/struct.Sender.html#method.send

> An unsuccessful send would be one where the corresponding receiver has already been deallocated

So there is only really error case, which does not necessarily indicate a failure mode